### PR TITLE
feat: share to twitter button

### PR DESCRIPTION
**Pull request by Agill-Sheron**
https://linear.app/geyser/issue/GYS-6744/update-share-project-to-include-share-to-twitter

### CODE REVIEW

**Review by steliosrammos — APPROVED** [review]

File: src/modules/project/pages/projectView/views/projectMainBody/sections/ShareProject.tsx [hunk 1]
```diff
@@ -36,6 +42,27 @@ export const ShareProject = () => {
...
+    const tweetKeys = [
+      'Check out my project in geyser: {{projectLink}}',
+      "We're on a mission to change the game and we need your help. Dive into our project on @geyserfund and let's make a difference. Are you in? {{projectLink}}",
+      "Dream with us! We're launching something incredible on @geyserfund and you're invited to be part of the journey. Join us! {{projectLink}}",
+      "Big news! We've just launched our latest project on @geyserfund and we need YOUR help to make it a reality. Ready to make a difference? {{projectLink}}",
+      "Join us in bringing our dream project to life on @geyserfund. Let's change things together! {{projectLink}}",
+      "Breaking news: We're shaking things up with our new project on @geyserfund! Check it out: {{projectLink}} #BreakingNews",
+      "We're transforming our dreams into reality with this project on @geyserfund. Your support matters: {{projectLink}}",
+      'Making a difference, one step at a time. Join our mission on @geyserfund and help us leave a lasting impact. Your support makes all the difference: {{projectLink}} #MakeADifference',
+    ]
```
Comment (sajald77) [inline src/modules/project/pages/projectView/views/projectMainBody/sections/ShareProject.tsx:57]: Might be better to get this const outside the component.

File: src/modules/project/pages/projectView/views/projectMainBody/sections/ShareProject.tsx [hunk 2]
```diff
@@ -36,6 +42,27 @@ export const ShareProject = () => {
...
+    const randomKey = tweetKeys[Math.floor(Math.random() * tweetKeys.length)] || ''
+    const tweet = t(randomKey)
+    const tweetWithProjectLink = tweet.replace('{{projectLink}}', projectLink)
+
+    const twitterUrl = `https://twitter.com/intent/tweet?text=${encodeURIComponent(tweetWithProjectLink)}`
+    window.open(twitterUrl, '_blank')
+  }
+
```
Comment (sajald77) [inline src/modules/project/pages/projectView/views/projectMainBody/sections/ShareProject.tsx:65]: Maybe create a reusable function for create tweet url. I can see us doing more of these in the future.

File: src/translations/English.json [hunk 3]
```diff
@@ -911,5 +912,12 @@
    "Choose Geyser split per transaction":"Choose Geyser split per transaction",
    "Lightning":"Lightning",
    "Onchain":"Onchain"
-
+   "Check out my project in geyser: {{projectLink}}":"Check out my project in geyser: {{projectLink}}",
```
Comment (sajald77) [inline src/translations/English.json:915]: Screenshot 2024-04-12 at 9 25 51 AM

It shows this is the reason build it failing, but I do see a comma, make sure the build passes.

File: src/modules/project/pages/projectView/views/projectMainBody/sections/ShareProject.tsx (hunk 2 above)
Comment (Agill-Sheron) [reply src/modules/project/pages/projectView/views/projectMainBody/sections/ShareProject.tsx:65]: Good point, would the `utils` folder be a good location for such function?

Comment (sajald77) [reply src/modules/project/pages/projectView/views/projectMainBody/sections/ShareProject.tsx:65]: eah, that's fair, for now, you could just create a utils.ts file closest to where it's used. If more such functions popup in the same level, we can turn it into a folder. Basically going for vertical split ( have files closest to where it's used).

File: src/modules/project/pages/projectView/views/projectMainBody/utils.tsx [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+export const shareOnTwitter = (tweetTemplates: string[], projectName: string, t: any) => {
+  const projectLink = `${window.location.origin}/project/${projectName}`
+
+  const randomKey = tweetTemplates[Math.floor(Math.random() * tweetTemplates.length)] || ''
+  const tweetTemplate = t(randomKey)
+  const twitterUrl = generateTwitterShareUrl(tweetTemplate, projectLink)
+
+  window.open(twitterUrl, '_blank')
+}
```
Comment (sajald77) [inline src/modules/project/pages/projectView/views/projectMainBody/utils.tsx:14]: last bit, the window.open doesnot work on certain privacy focused mobile browsers. it does not open it considering it as a popup, as it was triggered from JS instead by a user through a button click.

I suggest, move this code back to the button as an <a> tag, that opens a new window.

Comment (Agill-Sheron) [reply src/modules/project/pages/projectView/views/projectMainBody/utils.tsx:14]: on which browser did you test this so I can test it myself if it works?

**Review by sajald77 — APPROVED** [review]

**Pull request merged**